### ISSUE TITLE
Computation of the MID for the neurons on the sides

### ISSUE THREAD

**Issue by remidomingues** [JustGlowing/minisom#8]
Hi,

I'm using the interneuron distance matrix computed by the function distance_map, and saw in the comments of the function that the distance returned should be the normalized mean.

Yet, I don't see the computation of the mean of the distances. It seems that the code only sums up distances and then apply normalization. By doing so, the neurons located on the sides and having thus only 3 neighbors will have a lower distance (less additions and no averaging).

- Shouldn't we count the number of neighbors and divide the distance by this number?
- Shouldn't we ignore the distance between the neuron and itself in this computation?

To illustrate this, I attach a MID matrix that shows smaller (darker) distances on the sides, and the corresponding neural network in 2D (dimensionality reduction provided by MDS from scikit-learn).

![som_mid_qe 0 96](https://cloud.githubusercontent.com/assets/5238619/11179109/6f140d6e-8c51-11e5-9983-5d2d746ebdb9.png)

![som2d_grid true_data false_qe 0 96](https://cloud.githubusercontent.com/assets/5238619/11179111/7428220e-8c51-11e5-9ce4-3485b572b530.png)

After averaging, I obtain the following MID matrix:
![som_mid_qe 0 96](https://cloud.githubusercontent.com/assets/5238619/11179126/8f2cd6d0-8c51-11e5-9442-6d37bbc78dd6.png)

**Comment by JustGlowing** [comment]
Hi Remi,

As you noticed, `dinstance_map` simply returns the sum of the distances divided by the maximum. This partially avoids the problems at the border of the map and makes the computation a bit simpler.

Of course if you want to properly compute the average you have to divide by the correct number of neighbours and not take in account the comparisons between same neurons.

I hope this helps.
